### PR TITLE
Bump crd-docs-generator to 0.3.0 & add updated reference pages

### PR DESCRIPTION
**Pull request by ubergesundheit**
Towards giantswarm/giantswarm#15326

This PR bumps the version of crd-docs-generator to 0.3.0 which contains changes to the used template for rendering yaml code blocks.

It uses tripple backticks instead of `<pre><code>` markup. (Related giantswarm/crd-docs-generator#20)

It also deleted `storageconfigs.core.giantswarm.io`, `drainerconfigs.core.giantswarm.io` and `etcdbackups.backup.giantswarm.io` reference pages. Is this correct?

<details>

<summary>Screenshot</summary>

image

</details>

### CODE REVIEW

File: src/content/reference/management-api/drainerconfigs.core.giantswarm.io.md [hunk 1]
```diff
@@ -1,357 +0,0 @@
----
```
Comment (ubergesundheit) [inline src/content/reference/management-api/drainerconfigs.core.giantswarm.io.md:1]: Is this deletion (and others) ok?

Comment (stone-z) [reply src/content/reference/management-api/drainerconfigs.core.giantswarm.io.md:1]: Yes, these were intentionally excluded (drainerconfigs and storageconfigs are deprecated, etcdbackups are internal)

File: src/content/reference/management-api/appcatalogs.application.giantswarm.io.md [hunk 2]
```diff
@@ -37,7 +37,8 @@ aliases:
 
 
 <h3 id="crd-example-v1alpha1">Example CR</h3>
-<pre class="crd-example-cr"><code class="language-yaml">
```
Comment (stone-z) [inline src/content/reference/management-api/appcatalogs.application.giantswarm.io.md:40]: Is this class needed for anything else?

**Review by stone-z — APPROVED** [review]
This is nicer to work with anyway. If it renders fine then :rocket:

File: src/content/reference/management-api/drainerconfigs.core.giantswarm.io.md (hunk 1 above)
Comment (stone-z) [reply src/content/reference/management-api/drainerconfigs.core.giantswarm.io.md:1]: For reference: https://github.com/giantswarm/giantswarm/issues/15025

File: src/content/reference/management-api/appcatalogs.application.giantswarm.io.md (hunk 2 above)
Comment (ubergesundheit) [reply src/content/reference/management-api/appcatalogs.application.giantswarm.io.md:40]: Looks like it was only in the html, no styling done with these classes (`crd-example-cr` and `language-yaml`)

File: src/content/reference/management-api/drainerconfigs.core.giantswarm.io.md (hunk 1 above)
Comment (ubergesundheit) [reply src/content/reference/management-api/drainerconfigs.core.giantswarm.io.md:1]: Although to not break external links, we should consider keeping the pages around, clearly marking them as deprecated but not link them from the list.

**Comment by marians** [comment]
Thank you!

**Pull request merged**